upstairs/src/lib.rs

```diff
@@ -769,7 +774,7 @@ where
         gen: up.get_generation().await,
         read_only: up.read_only,
         encrypted: up.encrypted(),
-        supported_versions: vec![CRUCIBLE_MESSAGE_VERSION],
+        supported_versions: alternate_message_versions.clone(),
```

Comment:
maybe worth renaming this field since it means something slightly different now?